ROL/algorithm.cpp

```diff
@@ -10,29 +10,30 @@ namespace py = pybind11;
 #include <ROL_TypeB_QuasiNewtonAlgorithm.hpp>
 #include <ROL_Bounds.hpp>
 
+
 #include <fstream>
 #define FMT_HEADER_ONLY
 #include <fmt/core.h>
 
-void serialise_algorithm(const ROL::TypeB::Algorithm<double> &v, int rank) {
-  std::ofstream os(fmt::format("rol_typeb_algorithm_{}.cereal", rank), std::ios::binary);
+void serialise_algorithm(const ROL::TypeB::Algorithm<double> &v, int rank, std::string chdir="./") {
+  std::ofstream os(fmt::format("{0}/rol_typeb_algorithm_{1}.cereal", chdir, rank), std::ios::binary);
```

Comment:
This defaults to a path of `.//rol_typeb_algorithm_{rank}.cereal` ­— I think either `chdir` or the format string need to not have an extra slash. It could be like:
```cpp
std::filesystem::path checkpoint_dir(chdir);
... os(fmt::format(chdir / "rol_typeb_algorithm_{}.cereal", rank)
```
With a bit of massaging to get it to compile (maybe needs an explicit string comparison...) I guess not strictly necessary, but I prefer handling paths as an object rather than as strings.